keras/utils/timeseries_dataset.py

```diff
@@ -189,41 +208,21 @@ def timeseries_dataset_from_array(
   num_seqs = end_index - start_index - (sequence_length * sampling_rate) + 1
   if targets is not None:
     num_seqs = min(num_seqs, len(targets))
-  if num_seqs < 2147483647:
-    index_dtype = 'int32'
-  else:
-    index_dtype = 'int64'
-
-  # Generate start positions
-  start_positions = np.arange(0, num_seqs, sequence_stride, dtype=index_dtype)
-  if shuffle:
-    if seed is None:
-      seed = np.random.randint(1e6)
-    rng = np.random.RandomState(seed)
-    rng.shuffle(start_positions)
-
-  sequence_length = tf.cast(sequence_length, dtype=index_dtype)
-  sampling_rate = tf.cast(sampling_rate, dtype=index_dtype)
-
-  positions_ds = tf.data.Dataset.from_tensors(start_positions).repeat()
-
-  # For each initial window position, generates indices of the window elements
-  indices = tf.data.Dataset.zip(
-      (tf.data.Dataset.range(len(start_positions)), positions_ds)).map(
-          lambda i, positions: tf.range(  # pylint: disable=g-long-lambda
-              positions[i],
-              positions[i] + sequence_length * sampling_rate,
-              sampling_rate),
-          num_parallel_calls=tf.data.AUTOTUNE)
-
-  dataset = sequences_from_indices(data, indices, start_index, end_index)
+
+  dataset = tf.data.Dataset.from_tensor_slices(data[start_index : end_index])
+  dataset = dataset.window(sequence_length, shift=sequence_stride, stride=sampling_rate, drop_remainder=True)
```

Comment:
why are we forcing the drop_remainder to be True here?

Reply:
Because I need the same window size in all batches !!!

Reply:
@qlzh727 You have true only if I active `drop_remainder=True` at line 208 in `dataset = dataset.batch(batch_size) `. But this is another thing.

Reply:
I don't think the model need to have the same sequence length all the time. Its very common to have uneven sequence length dataset, and this is why the drop_remainder is a user configurable flag for tf.data API. See https://www.tensorflow.org/api_docs/python/tf/data/Dataset?version=nightly#window for the uneven example.

Reply:
@qlzh727  
Ok. I can make a `drop_remainder` flag to `timeseries_dataset_from_array` or close this PR whole.

Thanks.

Reply:
@qlzh727 the original code has same sequence length in whole dataset.

Look here: https://github.com/keras-team/keras/blob/master/keras/utils/timeseries_dataset.py#L215